pkg/util/winutil/pdhutil/pdhcounter.go

```diff
@@ -59,6 +59,9 @@ type PdhMultiInstanceCounterSet struct {
 // Initialize initializes a counter set object
 func (p *PdhCounterSet) Initialize(className string) error {
 
+	// refresh PDH object cache (refresh will only occur periodically)
+	CachedRefreshPdhObjectCache()
```

Comment:
Yes. It is only called on counter initialization. If all counters have already been successfully initialized then we no longer refresh.

Reply:
Go code has hardcoded counters (and specific instances with one exception). They often are already there. Which is different from Python code where it is setup in more dynamic arbitrary env and counters provider could be very dynamic too.

Reply:
> Maybe it worth to change function name to simply RefreshPdhObjectCache
I want a function name that indicates that the cache may not be refreshed when the function is called. I created `ForceRefreshPdhObjectCache` for instances where it may be desired to always refresh when called.